lib/netstd/Thrift/Transport/Server/TServerSocketTransport.cs

```diff
@@ -28,47 +28,27 @@ namespace Thrift.Transport.Server
     public class TServerSocketTransport : TServerTransport
     {
         private readonly int _clientTimeout;
-        private readonly int _port;
         private readonly bool _useBufferedSockets;
         private readonly bool _useFramedTransport;
         private TcpListener _server;
 
-        public TServerSocketTransport(TcpListener listener)
-            : this(listener, 0)
-        {
-        }
-
-        public TServerSocketTransport(TcpListener listener, int clientTimeout)
+        public TServerSocketTransport(TcpListener listener, int clientTimeout = 0, bool useBufferedSockets = false, bool useFramedTransport = false)
         {
             _server = listener;
             _clientTimeout = clientTimeout;
+            _useBufferedSockets = useBufferedSockets;
+            _useFramedTransport = useFramedTransport;
         }
 
-        public TServerSocketTransport(int port)
-            : this(port, 0)
-        {
-        }
-
-        public TServerSocketTransport(int port, int clientTimeout)
-            : this(port, clientTimeout, false)
-        {
-        }
-
-        public TServerSocketTransport(int port, int clientTimeout, bool useBufferedSockets):
-            this(port, clientTimeout, useBufferedSockets, false)
-        {
-        }
-        
-        public TServerSocketTransport(int port, int clientTimeout, bool useBufferedSockets, bool useFramedTransport)
+        public TServerSocketTransport(int port, int clientTimeout = 0, bool useBufferedSockets = false, bool useFramedTransport = false)
```

Comment:
It would have been cleaner to make a TcpListener here and then call the TcpListener constructor so there is one code path to initialization, not two.

Reply:
True, but you wouldn't be able to rethrow an exception as a TTransportException in this case. I'm not sure if that's a requirement though?

Reply:
Okay then, creating an initialize() method that does what the TcpListener does, and having both call it, would consolidate the code paths.  It isn't necessary, just a suggestion for future development.  Having two constructors initialize the same fields can lead to problems down the road when one constructor is modified and the other one is not.